pantab/src/reader.c

```diff
@@ -121,6 +121,8 @@ static PyObject *read_hyper_query(PyObject *Py_UNUSED(dummy), PyObject *args) {
     const size_t *sizes;
     const int8_t *null_flags;
 
+    PyDateTime_IMPORT;
```

Comment:
Can this be moved to the initialization function you have in pantab.c instead? It's usually included in an init rather than in a standard function

Reply:
unfortunately not... `PyDateTime_IMPORT` works in mysterious ways...

When importing `datetime.h`, it defines a
```
static PyDateTime_CAPI *PyDateTimeAPI = NULL;
```

This variable is `static` and thereby local to the c file into wich `datetime.h` was included.

This local variable is initialized by `PyDateTime_IMPORT` 
 (see https://github.com/python/cpython/blob/63298930fb531ba2bb4f23bc3b915dbf1e17e9e1/Include/datetime.h#L199)
 
 As we include `datetime.h` from both `reader.c` and `writer.c`, we need to call `PyDateTime_IMPORT` inside both our C files. Moving it to `pantab.c` won't work, as pantab.c has no access to the `static` variables from `reader.c` and `writer.c`
 
 I could introduce additional methods `reader_init` and `writer_init` which encapsulate the `PyDateTime_IMPORT` and could call those initialization functions from `pantab.c`.
 
 Performancewise, I am pretty sure calling `PyDateTime_IMPORT` once as part of every read/write is completely negligible.
 Also, the current solution is shorter/less lines of code.
 But if you prefer, I could of course introduce `reader_init` and `writer_init`.
 
 What are your thoughts?